sn_cli/src/subcommands/register.rs

```diff
@@ -43,15 +47,40 @@ pub(crate) async fn register_cmds(cmds: RegisterCmds, client: &Client) -> Result
     Ok(())
 }
 
+/// Crates a new register with the given pet name.
+/// Verifies it exists on the network before returning
 async fn create_register(name: String, client: &Client) -> Result<()> {
     let tag = 3006;
     let xorname = XorName::from_content(name.as_bytes());
     println!("Creating Register with '{name}' at xorname: {xorname:x} and tag {tag}");
 
     // clients currently only support public registers as we create a new key at each run
     let _register = ClientRegister::create_public_online(client.clone(), xorname, tag).await?;
-    println!("Successfully created register '{name}' at {xorname:?}, {tag}!");
-    Ok(())
+
+    let mut verification_attempts = 0;
+
+    while verification_attempts < VERIFICATION_ATTEMPTS {
+        match client.get_register(xorname, tag).await {
+            Ok(_) => {
+                println!("Successfully created register '{name}' at {xorname:?}, {tag}!");
```

Comment:
Being a bit pedantic here but `Ok(_)` here could well be another register created concurrently, we might want to check if it matches ours.

Reply:
Hmm, 
```
                assert_eq!(our_register.name(), returned_register.name());
```
is enough? comparing contents  might fail if we get _newer_ ?

Reply:
no, that'd be the same whatever for that get.... hmm. if we're doing this we need to tolerate potential _new_ entries

Reply:
Okay, i think I have a good enough way forwards here. Just checking the unferlying and if not matching erroring so the user can take action.